### PR TITLE
fixing react router dom version and switching to absolute routing whe…

### LINKED ISSUE #3830: Airlock: Creating an import/export request causes a routing error
**Describe the bug**
Upon creating a new export/import request in Airlock, the URL route malforms and causes the UI to behave in an weird way. The issue is to do with routing when creating a new request. I have already forked the repo and fixed the issue. The issue to be exact is here: https://github.com/microsoft/AzureTRE/blob/31960894e15266f6ae76a4b7c1028cf9cc6bc459/ui/app/src/components/shared/airlock/Airlock.tsx#L258 The path is being navigated from a relative point of view and just appended rather then re-written. The fix I put in the fork is the following:
```typescript
    navigate(`../${newRequest.id}`, { relative: "path" });
``` 

**Steps to reproduce**

1. Open/create a new workspace and navigate to **Airlock** on the side panel.
2. Create a new import or export request.
3. Fill in the form and press create.

Expected:
The draft request to be created and the side panel to open automatically with the SAS URL information, with the URL `/requests/{id}`.

Actual:
The side panel closes and the path remains `/requests/new/{id}` but the side panel doesn't reopen with the SAS URL information. Double clicking requests simply appends the airlock id to the URL.

Work around: 
Exit and enter back to the requests list view and double clicking from there allows the side panel to open.

**Azure TRE release version**: v0.15.2, **UI version**: 0.5.17, (but can also see the issue on main).

### PR DESCRIPTION
**Pull request by yahya130**
Fixes #3830  

This PR fixes the following:

- Nails down the react-dom-router dependency to version 6.21.1.
- Updates the routing path to be more absolute when creating an airlock request.

### CODE REVIEW

**Comment by marrobi** [comment]
Thanks @yahya130 

@tamirkamara any idea why `flag_external_pr` is failing .

Could it be:
> This org has transitioned to a read-only GITHUB_TOKEN for GitHub Action workflows. This does break your builds. See https://aka.ms/github-token-perms-changes for how to resolve.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 <!-- markdownlint-disable MD041 -->
 ## 0.17.0 (Unreleased)
 
+BUG FIXES:
```
Comment (marrobi) [inline CHANGELOG.md:4]: Looks like we have two lots of bug fixes for this release, can you add lower down. Thanks.

**Comment by marrobi** [comment]
> Thanks @yahya130
> 
> @tamirkamara any idea why `flag_external_pr` is failing .
> 
> Could it be:
> 
> > This org has transitioned to a read-only GITHUB_TOKEN for GitHub Action workflows. This does break your builds. See https://aka.ms/github-token-perms-changes for how to resolve.

Actually can see the GitHub Scripts action uses GITHUB_TOKEN.  Will have to look to het the workflows fixed.

**Comment by tamirkamara** [comment]
@marrobi it is this. I'm adding a permissions block in a pr.

> > any idea why `flag_external_pr` is failing .
> > Could it be:
> > > This org has transitioned to a read-only GITHUB_TOKEN for GitHub Action workflows. This does break your builds. See https://aka.ms/github-token-perms-changes for how to resolve.
> 
> Actually can see the GitHub Scripts action uses GITHUB_TOKEN. Will have to look to het the workflows fixed.

**Review by jjgriff93 — APPROVED** [review]

**Comment by benc-uk** [comment]
Random (ish) stranger here! 👋
Just a suggestion - it's normally good practice to add the update the updated package-lock.json whenever the package.json is updated.
And actually changes to dependency versions in the package.json might not even take place until the lockfile is updated too

**Comment by yahya130** [comment]
> Random (ish) stranger here! 👋 Just a suggestion - it's normally good practice to add the update the updated package-lock.json whenever the package.json is updated. And actually changes to dependency versions in the package.json might not even take place until the lockfile is updated too

Hi @benc-uk  👋, not random-ish at all 😄 great suggestion and well noted, just to add more context, the app uses yarn as the package manager and the package.json file has been updated to reflect what it is in the yarn.lock file.

**Comment by marrobi** [comment]
@yahya130 @benc-uk should there be an updated `yarn.lock` file then? There is one in the repo, but not this PR.

**Comment by yahya130** [comment]
> @yahya130 @benc-uk should there be an updated `yarn.lock` file then? There is one in the repo, but not this PR.

It looks ok:
https://github.com/microsoft/AzureTRE/blob/31960894e15266f6ae76a4b7c1028cf9cc6bc459/ui/app/yarn.lock#L9395-L9408

@benc-uk what would be your opinion on this?

**Comment by benc-uk** [comment]
Totally missed you're using yarn
Yep that looks good!

**Comment by marrobi** [comment]
@tamirkamara looks like the bot is fixed. Thanks.

@yahya130 As this is UI and doesn't have tests, but will let it run through so we can ensure our builds are fixed, then merge. Thanks!

**Comment by marrobi** [comment]
Looks like our tests are blocked by - https://github.com/getporter/porter/issues/2987

Will force approval given this only touches UI. But the bot seems to be fixed.

**Comment by marrobi** [comment]
@tamirkamara seems still an issue, tests aren't being marked as complete. Looks like granted the right permissions here - https://github.com/microsoft/AzureTRE/blob/3835e5ad89209d55c131eaa8e6b4211fbcc6c9ff/.github/workflows/pr_comment_bot.yml#L21 , can't see any errors either.

Any ideas?

**Pull request merged**